catalog/spatial/geocoding/spatial-geocoding-plugin/src/main/java/org/codice/ddf/spatial/geocoding/plugin/GeoCoderPlugin.java

```diff
@@ -118,11 +106,38 @@ public int getRadius() {
         return radiusInKm;
     }
 
-    private boolean countryCodeNeeded(Metacard metacard) {
-        return metacard.getAttribute(Location.COUNTRY_CODE) == null;
+    /**
+     * Sets the country code attribute of the {@param metacard} when the {@param metacard} has a
+     * location and there is not already a country code attribute. Does not update the country code
+     * if the country code attribute is already set.
+     *
+     * @param metacard
+     */
+    private void setCountryCode(Metacard metacard) {
+        Optional<String> wktLocation = Optional.ofNullable(metacard.getLocation());
+
+        if (wktLocation.isPresent() && !hasCountryCode(metacard)) {
+            GeoCoder geocoder = geoCoderFactory.getService();
```

Comment:
Is there any reason we need to get the GeoCoder for each metacard? I would think obtaining to once for the entire list would be more efficient.